rust/datafusion/src/execution/aggregate.rs

```diff
@@ -312,6 +313,80 @@ impl AggregateFunction for SumFunction {
     }
 }
 
+/// Implemntation of COUNT aggregate function
+#[derive(Debug)]
+struct CountFunction {
+    data_type: DataType,
```

Comment:
we don't need to track a data type for count, since it should always be `usize`

Reply:
We do need a `DataType` though if the `AggregateFunction` trait has to be implement for `CountFunction`, don't we?
I could avoid implementing the `AggregateFunction` trait for `CountFunction`, but then I don't see how we could use `CountFunction` in the context of `AccumulatorSet`.

Reply:
The data type of count should always be `usize`. Yes, we still need to implement the `AggregateFunction` trait, but the return_type method can be hard-coded as returning `DataType::UInt32`. The data type of count is not related to the data type of the things being counted.